src/smc-webapp/course/store.coffee

```diff
@@ -274,7 +274,7 @@ exports.CourseStore = class CourseStore extends Store
         return x.get('time')
 
     has_grade: (assignment, student_id) =>
-        return @get_assignment(assignment)?.get("grades")?.get(student_id)
+        return !!assignment.get("grades")?.get(student_id)
```

Comment:
oddly enough, this was always wrong, because `assignment` is here already the assignment object (not a key). Given it's name, it should also be a boolean.

Reply:
Sorry, but change this back!  

It's not broken, it's not wrong, and it is not always wrong.  Everything that takes a var called "assignment" is designed to also work with just an `assignment_id` as input instead.  That's the reason for `@get_assignment` -- it just returns it if already an assignment, otherwise fetches the assignment from the id.